core/jobs/src/main/java/com/cognifide/aet/job/common/modifiers/login/LoginModifier.java

```diff
@@ -135,35 +78,46 @@ public CollectorStepResult collect() throws ProcessingException {
     return CollectorStepResult.newModifierResult();
   }
 
-  @Override
-  public void setParameters(Map<String, String> params) throws ParametersException {
-    loginPage = params.get(LOGIN_PAGE_PARAM);
-    login = StringUtils.defaultString(params.get(LOGIN_PARAM), DEFAULT_LOGIN);
-    password = StringUtils.defaultString(params.get(PASSWORD_PARAM), DEFAULT_PASSWORD);
-
-    loginInputSelector = StringUtils.defaultString(params.get(LOGIN_INPUT_SELECTOR_PARAM),
-            DEFAULT_LOGIN_INPUT_ELEMENT_SELECTOR);
-    passwordInputSelector = StringUtils.defaultString(params.get(PASSWORD_INPUT_SELECTOR_PARAM),
-            DEFAULT_PASSWORD_INPUT_ELEMENT_SELECTOR);
-    submitButtonSelector = StringUtils.defaultString(params.get(SUBMIT_BUTTON_SELECTOR_PARAM),
-            DEFAULT_SUBMIT_BUTTON_ELEMENT_SELECTOR);
-
-    loginTokenKey = StringUtils.defaultString(params.get(LOGIN_TOKEN_KEY_PARAM), DEFAULT_LOGIN_TOKEN);
-    final String timeoutString = params.get(LOGIN_CHECK_TIMEOUT_PARAM);
-
-    ParametersValidator.checkNotBlank(loginPage, "`login-page` parameter is mandatory");
-    if (StringUtils.isNotBlank(timeoutString)) {
-      loginCheckTimeout = NumberUtils.toInt(timeoutString);
-      ParametersValidator.checkRange(loginCheckTimeout, 0, 10000, "Timeout duration should be greater than 0 and " +
-              "less than 10 seconds");
+  private void login() throws ProcessingException {
+    loginToForm();
+    sleep();
+
+    Cookie authCookie = getLoginToken();
+    if (authCookie == null) {
+      throw new ProcessingException("Unable to acquire Cookie; check credentials.");
     }
-    if (params.containsKey(FORCE_LOGIN)) {
-      this.forceLogin = Boolean.valueOf(params.get(FORCE_LOGIN));
+
+    webCommunicationWrapper.getHttpRequestBuilder().addCookie(authCookie.getName(), authCookie.getValue());
+    LOGGER.info("User has been authenticated");
+  }
+
+  private void loginToForm() throws ProcessingException {
+    LOGGER.info("Attempting to login user: {} on site {}", config.getLogin(), config.getLoginPage());
+    webDriver.get(config.getLoginPage());
+    LoginFormComponent form = new LoginFormComponent(webDriver, config.getLoginInputSelector(),
+            config.getPasswordInputSelector(), config.getSubmitButtonSelector());
+    form.login(config.getLogin(), config.getPassword());
+  }
+
+  private void sleep() {
+    if (config.getLoginCheckTimeout() > 0) {
+      LOGGER.info("Waiting {} ms.", config.getLoginCheckTimeout());
+      try {
+        Thread.sleep(config.getLoginCheckTimeout());
+      } catch (InterruptedException e) {
+        LOGGER.error("Interruption", e);
+        Thread.currentThread().interrupt();
```

Comment:
I think this is so little damage here when `sleep` is interrupted, we can go with this approach. It looks that `SleepModifier` handles interruption in similar way - it may be area to improve at least logging there.